### ISSUE TITLE
Cloudflare Workers

### ISSUE THREAD

**Issue by evgenydmitriev** [ccxt/ccxt#19125]
### Programming Languages

JavaScript

### Description

Hello! [This part](https://github.com/ccxt/ccxt/blob/e634e94fa940a2ca76c98d2ffc1aac65e143fa22/ts/src/base/Exchange.ts#L910-L919) is blocking deployments in Cloudflare Workers. Would you consider an environment check and a switch to browser's `fetch` instead of node's `http` to address the node dependencies not provided by [Cloudflare Runtime APIs](https://developers.cloudflare.com/workers/runtime-apis/nodejs)? Otherwise, users need to maintain their builds with isNode set to false.

### Code

```
  

```

**Comment by sc0Vu** [comment]
Hi @evgenydmitriev

I think nodejs support `node:` protocol after 14.18 or some version. Would you like to check which nodejs version on the machine?

**Comment by evgenydmitriev** [comment]
> Uh, looks like cloudflare didn't support http in node runtime......

Exactly, we got it working by [switching to `fetch`](https://developers.cloudflare.com/workers/runtime-apis/fetch/) manually.

**Comment by carlotestor** [comment]
Good news — this is resolved on current CCXT. The base `Exchange` no longer hard-requires Node's `http`/`https` modules: `loadFetchImplementation()` now detects the runtime and uses the platform-native `fetch` in browsers, Deno, Bun and Workers, only reaching for undici under real Node. So Cloudflare Workers deployments no longer need `isNode = false` builds or a manual `fetch` swap.

Just upgrade:
```bash
npm i ccxt@latest
```

and the plain import should build and deploy:
```js
import ccxt from 'ccxt';

export default {
    async fetch (request, env, ctx) {
        const exchange = new ccxt.binance ();
        const ticker = await exchange.fetchTicker ('BTC/USDT');
        return new Response (JSON.stringify (ticker));
    },
};
```

If you still hit a node-builtin resolution error in Workers, add `compatibility_flags = ["nodejs_compat"]` to your `wrangler.toml` — that covers the remaining optional deps (proxy agents, ws) that are lazily imported and never touched on the Workers path.

Thanks for the original report — glad your manual `fetch` workaround got you unblocked in the meantime.
